### PR TITLE
feat #18: develop header component

### PR DESCRIPTION
**Pull request by sedaklnc**
## What has been done 
- [x] Responsive design
- [x] Dynamic props
- [x] Typing props

## What will be done?
- [x] Binding auth process to props
- [x] Developing drop-down menus

## What is missing?
- Prettier and eslint configuration do not match
- Dropdown menus do not have figma drawings
- Auth function not improved

## Who developed?
- Seda @sedaklnc
- Sami @ssibrahimbas

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
 /** @type {import('next').NextConfig} */
 const nextConfig = {
   reactStrictMode: true,
-  swcMinify: true
+  swcMinify: true,
+  images: { domains: ['prototyprio.gumlet.io'] }
```
Comment (mehmetsagir) [inline next.config.js:5]: The recommended domain addition method in configuration with Next 13 should be as follows. **[Remote Patterns](https://beta.nextjs.org/docs/optimizing/images#defining-domains-for-remote-images)**

File: components/Header/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+            <X onClick={() => setIsOpen(false)} />
+          ) : (
+            <Menu onClick={() => setIsOpen(true)} />
+          )}
+        </div>
+      </div>
+
+      {isOpen ? (
+        <>
+          <div className="flex  md:p-0 md:w-max p-4  grow w-full items-start md:items-center justify-start">
```
Comment (wralith) [inline components/Header/Header.tsx:92]: There are formatting issues in className strings -double or triple spaces- here and there. Would be easy to fix after #40 i guess.

Comment (ssibrahimbas) [reply components/Header/Header.tsx:92]: yes, thanks. Prettier and eslint did not match when we were developing this component. So we added this to the "What is missed" field when opening the merge-request. we will update.

File: components/Header/Header.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+export default function Header({
+  imageUrl = 'https://prototyprio.gumlet.io/strapi/20b5e1615519f85953d636ec6a5005ae.png?w=3840&q=75&format=webp&compress=true&dpr=2',
+  title = 'Untitled UI',
+  topMenuItems = defaultTopMenuItems,
+  menuItems = defaultMenuItems
+}: HeaderParams) {
+  const [isOpen, setIsOpen] = useState(true);
+
+  useEffect(() => {
+    if (typeof window !== 'undefined' && window.innerWidth < 768) {
```
Comment (eraygundogmus) [inline components/Header/Header.tsx:70]: bu condition'a güzel bir naming yapabiliriz. 
örneğin: 
isClientInMobileView = typeof window !== 'undefined' && window.innerWidth < 768

if (isClientInMobileView)

File: components/Header/Header.tsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+            <button className="px-2 w-full md:w-max py-1 rounded hover:bg-slate-100 opacity-70 ease-in-out duration-300">
+              Log in
+            </button>
+            <button className="bg-indigo-500 md:w-max w-full hover:bg-indigo-600 text-white px-2 py-1 rounded ease-in-out duration-300">
+              Sign up
+            </button>
+          </div>
+        </>
+      ) : (
+        <></>
```
Comment (eraygundogmus) [inline components/Header/Header.tsx:120]: Bunun anlamı nedir tam olarak? False return etmemek için mi?

File: components/Header/Header.tsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+          ) : (
+            <Menu onClick={() => setIsOpen(true)} />
+          )}
+        </div>
+      </div>
+
+      {isOpen ? (
+        <>
+          <div className="flex  md:p-0 md:w-max p-4  grow w-full items-start md:items-center justify-start">
+            <div className="flex w-full flex-col md:flex-row md:items-center items-start md:gap-x-6 md:gap-y-0 gap-6">
```
[the remark this answers is not in the document]
Comment (ssibrahimbas) [reply components/Header/Header.tsx:93]: kodları yazdığımız sırada eslint ve prettier konfigürasyonu tamamlanmamıştı. Bugün main branch'ından güncelledikten sonra lint'i çalıştıracağız

File: components/Header/Header.tsx (hunk 4 above)
Comment (ssibrahimbas) [reply components/Header/Header.tsx:120]: Evet false durumunda return etmemek için bu şekilde yaptık. Sanırım && ile yazınca aynı sonucu veriyor. Güncelleyeceğiz

File: components/Header/Header.tsx [hunk 6]
```diff
@@ -0,0 +1,124 @@
+import Image from 'next/image';
+import React, { useState, useEffect } from 'react';
+import { Menu, X } from 'react-feather';
+
+type MenuItem = {
+  text: string;
+  path?: string;
+};
```
Comment (eraygundogmus) [inline components/Header/Header.tsx:8]: can we split this into a different .ts file for types?

File: components/Header/Header.tsx [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+const defaultMenuItems: MenuItem[] = [
```
Comment (eraygundogmus) [inline components/Header/Header.tsx:9]: I think, this menu items should stay in some global store. Not in a jsx file

File: components/Header/Header.tsx [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+  path?: string;
+};
+type HeaderParams = {
+  imageUrl?: string;
+  title?: string;
+  topMenuItems?: TopMenuItem[];
+  menuItems?: MenuItem[];
+};
+
+const defaultTopMenuItems: TopMenuItem[] = [
```
Comment (eraygundogmus) [inline components/Header/Header.tsx:47]: same for here

**Review by eraygundogmus — CHANGES_REQUESTED** [review]

File: components/Header/Header.tsx (hunk 2 above)
Comment (ssibrahimbas) [reply components/Header/Header.tsx:92]: Should we use the semantic button element or wait for the developed button element?

**Comment by sedaklnc** [comment]
## What has been done
- [x] run format and lint scripts
- [x] update remote pattern for nextjs13
- [x] add discord button instead of auth buttons
- [x] add semantic button for mobile icons
- [x] reformat code for better intelligibility

## What will Be done?
- [ ] when the button component is completed, it will be implemented

## What is missing?
- [ ] Menus will be put into global scope: Waiting for team decision

File: components/Header/Header.tsx (hunk 2 above)
Comment (oguzergul) [reply components/Header/Header.tsx]: nope. this is not about our button component. you need to use the native button element `<button>`

**Comment by ssibrahimbas** [comment]
## Things to do after live review

- [ ] The brekpoint function in useEffect will be removed.
- [ ] Menu items will be added without prop
- [ ] It will be Title Frontendship.
- [ ] Semantic HTML elements will be used.
- [ ] Links will be replaced with next/link in Link elements

**Review by eraygundogmus — APPROVED** [review]

**Comment by fdemir** [comment]
We should generate the header items at build time. It is possible by using getStaticProps. No need to save the menu items in memory(store).

**Review by oguzergul — CHANGES_REQUESTED** [review]

File: components/Header/Header.tsx [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+  );
+};
+
+const SubMenu: React.FC<SubMenuProps> = ({ columns }) => {
+  return (
+    <nav className="flex w-full justify-between pt-4">
+      <div className="flex flex-row gap-3 pl-4">
+        {columns.map((col, index) => (
+          <div className="flex flex-col gap-3" key={index}>
+            <h3 className="text-indigo-500  ">{col.title}</h3>
```
[the remark this answers is not in the document]
Comment (ssibrahimbas) [reply components/Header/Header.tsx:95]: We have to do this because it is opened by clicking the header button on mobile. It cannot be done with css.

**Pull request merged**